### PR TITLE
[IMPROVEMENT] Improved Logging

### PR DESCRIPTION
**Pull request by underscoreorcus**
Logging improved in read_wav_file.cpp using the in-built logging function.

### CODE REVIEW

File: src/lib_ccaligner/read_wav_file.cpp [hunk 1]
```diff
@@ -105,6 +105,7 @@ bool WaveFileData::decode()     //decodes the wave file
     if(format != "WAVE")
     {
         FATAL(EXIT_INVALID_FILE, "Invalid WAV file format : %s ", format.c_str());
+        LOG("File format couldn't be identified as WAV");
```
Comment (saurabhshri) [inline src/lib_ccaligner/read_wav_file.cpp]: After FATAL, the function won't continue further and hence this LOG will never be executed :)

**Comment by saurabhshri** [comment]
@underscoreorcus See my comment. Please fix it at every place

File: src/lib_ccaligner/read_wav_file.cpp (hunk 1 above)
Comment (underscoreorcus) [reply src/lib_ccaligner/read_wav_file.cpp]: Oh, damn! I should have noticed this. I'll do the changes!

File: src/lib_ccaligner/read_wav_file.cpp [hunk 2]
```diff
@@ -139,15 +137,13 @@ bool WaveFileData::decode()     //decodes the wave file
...
-        LOG("Invalid SubChunk1ID");
     }
 
     unsigned long subChunk1Size = fourBytesToInt(_fileData, fmtIndex + 4);
 
     if(subChunk1Size != 16)
     {
-        FATAL(EXIT_INVALID_FILE, "Not PCM, SubChunk1Size : %lu", subChunk1Size);
-        LOG("SubChunk1Size not 16");
+        FATAL(EXIT_INVALID_FILE, "Not PCM, SubChunk1Size : %lu", subChunk1Size);;
```
Comment (saurabhshri) [inline src/lib_ccaligner/read_wav_file.cpp]: There are two semi-colons in this line.

File: src/lib_ccaligner/read_wav_file.cpp [hunk 3]
```diff
@@ -623,11 +619,14 @@ bool WaveFileData::read()   //decided the function based on set mode
...
+        case readFile             : LOG("Opening mode chosen, proceeding")
+                                    openFile();             //file on disk
                                     break;
-        case readStreamDirectly   : readStream();           //from stream/pipe
+        case readStreamDirectly   : LOG("Opening mode chosen, proceeding")
+                                    readStream();           //from stream/pipe
                                     break;
-        case readStreamIntoBuffer : readStreamUsingBuffer();//from stream/pipe into buffer and then processing
+        case readStreamIntoBuffer : LOG("Opening mode chosen, proceeding")
+                                    readStreamUsingBuffer();//from stream/pipe into buffer and then processing
```
Comment (saurabhshri) [inline src/lib_ccaligner/read_wav_file.cpp]: All these messages look same, maybe mention the mode chosen.

File: src/lib_ccaligner/read_wav_file.cpp [hunk 4]
```diff
@@ -186,23 +182,20 @@ bool WaveFileData::decode()     //decodes the wave file
...
         FATAL(EXIT_INVALID_FILE, "Incorrect header, ByteRate and/or BlockAlign values do not match!");
-        LOG("Error: Incorrect header");
     }
 
     std::string subChunk2ID (_fileData.begin() + dataIndex, _fileData.begin() + dataIndex + 4);
 
     if(subChunk2ID != "data")
     {
         FATAL(EXIT_INVALID_FILE, "Invalid SubChunk2ID : %s", subChunk2ID.c_str());
-        LOG("Invalid SubChunk2ID");
```
Comment (saurabhshri) [inline src/lib_ccaligner/read_wav_file.cpp]: You can shift them above FATAL as well.

**Pull request merged**